python/python/lance/sampler.py

```diff
@@ -69,15 +69,15 @@ def _efficient_sample(
     chunk_sample_size = n // max_takes
     for idx, i in enumerate(range(0, total_records, chunk_size)):
         # Add more randomness within each chunk.
-        offset = i + np.random.randint(0, chunk_size - chunk_sample_size)
+        offset = min(total_records - chunk_sample_size, i + np.random.randint(0, chunk_size - chunk_sample_size))
```

Comment:
I wonder if the original `offset` should have rather been `offset = idx + np.random.randint(0, chunk_size - chunk_sample_size)`? (`i` enumerates chunks, so `idx == i * chunk_size`).
```suggestion
        offset = min(total_records - chunk_sample_size, idx + np.random.randint(0, chunk_size - chunk_sample_size))
```

Reply:
Sorry, I misread the for loop statement. This is ok as you propose it.